packages/itwinui-react/src/core/ColorPicker/ColorBuilder.tsx

```diff
@@ -253,37 +265,49 @@ export const ColorBuilder = React.forwardRef((props, ref) => {
       {...rest}
     >
       <Box
-        className='iui-color-field'
+        as='div'
+        {...colorFieldProps}
+        className={cx('iui-color-field', colorFieldProps?.className)}
         style={
           {
             '--iui-color-field-hue': hueColorString,
             '--iui-color-picker-selected-color': dotColorString,
+            ...colorFieldProps?.style,
           } as React.CSSProperties
         }
-        ref={squareRef}
-        onPointerDown={(event: React.PointerEvent) => {
-          event.preventDefault();
-          updateSquareValue(event, 'onClick');
-          setColorDotActive(true);
-          colorDotRef.current?.focus();
-        }}
+        ref={useMergedRefs(squareRef, colorFieldProps?.ref)}
+        onPointerDown={mergeEventHandlers(
+          (event: React.PointerEvent) => {
+            event.preventDefault();
+            updateSquareValue(event, 'onClick');
+            setColorDotActive(true);
+            colorDotRef.current?.focus();
+          },
+          colorFieldProps?.onPointerDown,
```

Comment:
the event handler from props should be executed first, not last

same comment applies in all other instances of `mergeEventHandlers`

Reply:
I have reordered `mergeEventHandlers`.